src/Umbraco.Infrastructure/Security/MembersUserStore.cs

```diff
@@ -0,0 +1,703 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Umbraco.Core;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Identity;
+using Umbraco.Core.Scoping;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Infrastructure.Security
+{
+    /// <summary>
+    /// A custom user store that uses Umbraco member data
+    /// </summary>
+    public class MembersUserStore : UserStoreBase<MembersIdentityUser, IdentityRole<string>, string, IdentityUserClaim<string>, IdentityUserRole<string>, IdentityUserLogin<string>, IdentityUserToken<string>, IdentityRoleClaim<string>>
+    {
+        private readonly IMemberService _memberService;
+        private readonly UmbracoMapper _mapper;
+        private readonly IScopeProvider _scopeProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MembersUserStore"/> class for the members identity store
+        /// </summary>
+        /// <param name="memberService">The member service</param>
+        /// <param name="mapper">The mapper for properties</param>
+        /// <param name="scopeProvider">The scope provider</param>
+        /// <param name="describer">The error describer</param>
+        /// 
+        public MembersUserStore(IMemberService memberService, UmbracoMapper mapper, IScopeProvider scopeProvider, IdentityErrorDescriber describer)
+        : base(describer)
+        {
+            _memberService = memberService ?? throw new ArgumentNullException(nameof(memberService));
+            _mapper = mapper;
+            _scopeProvider = scopeProvider;
+        }
+
+        /// <summary>
+        /// Not supported in Umbraco
+        /// </summary>
+        /// <inheritdoc />
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override IQueryable<MembersIdentityUser> Users => throw new NotImplementedException();
```

Comment:
Interesting. I'll have a think/chat and get back to you!

The shared UserManager base class actually is already doing the second suggestion of overriding this property, for example:
 
`public override bool SupportsQueryableUsers => false; `
 
So are you saying we could override all the "Supports" properties on the shared `UserManager` to explain what our user stores support? e.g. SupportsUserTwoFactorRecoveryCodes, SupportsUserPhoneNumber etc set to false?
 
I'm not sure re no longer using the Microsoft identity UserStoreBase class - even if it means we have to specify what we don't support - will speak to the team tomorrow morning and let you know what we think.

Reply:
@Shazwazza I had a chat with Scott and we're a bit torn, but our thoughts are:

- Having all these overridden methods in the User Manager might feel a little cumbersome. However, we're already doing this in the shared base `UmbracoUserManager` class so each user manager would override with their own requirements if they differed.
- We could move away from our member user store(s) deriving from the Microsoft identity `UserStoreBase` class,
but then we end up implementing and having to maintain our own pattern. We'd also lose out on future potential updates to the Microsoft library.

Overall, I think we'll leave it in this PR as it is now, but something to consider especially if we're sharing more code between the Users and Members implementation. I think this could qualify for a separate GitHub issue to discuss approach.